src/main/java/org/broadinstitute/hellbender/tools/walkers/GenotypeGVCFs.java

```diff
@@ -0,0 +1,334 @@
+        attrs.put(GATKVCFConstants.MLE_ALLELE_FREQUENCY_KEY, newVC.getAttribute(GATKVCFConstants.MLE_ALLELE_FREQUENCY_KEY));
+        if (newVC.hasAttribute(GATKVCFConstants.NUMBER_OF_DISCOVERED_ALLELES_KEY))
+            attrs.put(GATKVCFConstants.NUMBER_OF_DISCOVERED_ALLELES_KEY, newVC.getAttribute(GATKVCFConstants.NUMBER_OF_DISCOVERED_ALLELES_KEY));
+
+        return new VariantContextBuilder(newVC).attributes(attrs).make();
+    }
+
+
+    /**
+     * Cleans up genotype-level annotations that need to be updated.
+     * 1. move MIN_DP to DP if present
+     * 2. propagate DP to AD if not present
+     * 3. remove SB if present
+     * 4. change the PGT value from "0|1" to "1|1" for homozygous variant genotypes
+     * 5. move GQ to RGQ if the site is monomorphic
+     *
+     * @param VC            the VariantContext with the Genotypes to fix
+     * @param createRefGTs  if true we will also create proper hom ref genotypes since we assume the site is monomorphic
+     * @return a new set of Genotypes
+     */
+    private List<Genotype> cleanupGenotypeAnnotations(final VariantContext VC, final boolean createRefGTs) {
+        final GenotypesContext oldGTs = VC.getGenotypes();
+        final List<Genotype> recoveredGs = new ArrayList<>(oldGTs.size());
+        for ( final Genotype oldGT : oldGTs ) {
+            final Map<String, Object> attrs = new HashMap<>(oldGT.getExtendedAttributes());
+
+            final GenotypeBuilder builder = new GenotypeBuilder(oldGT);
+            int depth = oldGT.hasDP() ? oldGT.getDP() : 0;
+
+            // move the MIN_DP to DP
+            if ( oldGT.hasExtendedAttribute("MIN_DP") ) {
+                depth = Integer.parseInt((String)oldGT.getAnyAttribute("MIN_DP"));
+                builder.DP(depth);
+                attrs.remove("MIN_DP");
+            }
+
+            // move the GQ to RGQ
+            if ( createRefGTs && oldGT.hasGQ() ) {
+                builder.noGQ();
+                attrs.put(GATKVCFConstants.REFERENCE_GENOTYPE_QUALITY, oldGT.getGQ());
+            }
+
+            // remove SB
+            attrs.remove("SB");
+
+            // update PGT for hom vars
+            if ( oldGT.isHomVar() && oldGT.hasExtendedAttribute(GATKVCFConstants.HAPLOTYPE_CALLER_PHASING_GT_KEY) ) {
+                attrs.put(GATKVCFConstants.HAPLOTYPE_CALLER_PHASING_GT_KEY, "1|1");
+            }
+
+            // create AD if it's not there
+            if ( !oldGT.hasAD() && VC.isVariant() ) {
+                final int[] AD = new int[VC.getNAlleles()];
+                AD[0] = depth;
+                builder.AD(AD);
+            }
+
+            if ( createRefGTs ) {
+                final int ploidy = oldGT.getPloidy();
+                final List<Allele> refAlleles = Collections.nCopies(ploidy,VC.getReference());
+
+                //keep 0 depth samples as no-call
+                if (depth > 0) {
```

Comment:
put some of the code from above inside of this `if`. It''ll make it less painful to see those dead temps